plugins/languages/treesitter/treesitter.nix

```diff
@@ -12,6 +12,101 @@ helpers.neovim-plugin.mkNeovimPlugin config {
   luaName = "nvim-treesitter.configs";
   defaultPackage = pkgs.vimPlugins.nvim-treesitter;
 
+  description = ''
+    Provides an interface to [tree-sitter]
+
+    ## Installing Your Own Grammars with Nixvim
+
+    The `nvim-treesitter` derivation generates the grammar packages based on the
+    default configs of nvim-treesitter but you can always install your own
+    grammars.
+
+    ```nix
+    { pkgs, ... }:
+    let
+      #example of adding your own grammar:
+      treesitter-nu-grammar = pkgs.tree-sitter.buildGrammar {
+        language = "nu";
+        version = "0.0.0+rev=0bb9a60";
+        src = pkgs.fetchFromGitHub {
+          owner = "nushell";
+          repo = "tree-sitter-nu";
+          rev = "0bb9a602d9bc94b66fab96ce51d46a5a227ab76c";
+          hash = "sha256-A5GiOpITOv3H0wytCv6t43buQ8IzxEXrk3gTlOrO0K0=";
+        };
+        meta.homepage = "https://github.com/nushell/tree-sitter-nu";
+      };
+
+      # or you can yoink any grammars in tree-sitter.grammars.$\{grammar-name}
+      # treesitter-nu-grammar = pkgs.tree-sitter-grammars.tree-sitter-nu;
+    in
+    {
+
+      programs.nixvim.plugins = {
+        treesitter = {
+          enable = true;
+          indent = true;
+          grammarPackages = pkgs.vimPlugins.nvim-treesitter.passthru.allGrammars ++ [
+            treesitter-nu-grammar
+          ];
+        };
+      };
+
+      programs.nixvim.extraConfigLua =
+        '''
+          do
+            local parser_config = require "nvim-treesitter.parsers".get_parser_configs()
+            -- change the following as needed
+            parser_config.nu = {
+              install_info = {
+                url = "''${treesitter-nu-grammar}", -- local path or git repo
```

Comment:
Did you test this out? From my quick experimenting and following their [readme](https://github.com/nvim-treesitter/nvim-treesitter/?tab=readme-ov-file#adding-parsers) this path needs to find the files below but you link to the derivation above.

Reply:
Yes. These two examples are actually from my config.

Reply:
This section is used for `:TSInstall nu`. But, you install it below with `extraPlugins`. If you ran `:TSInstall nu` you should see a message about a parser not managed by treesitter.

Reply:
I was able follow the example and set up [fsharp](https://github.com/khaneliman/astronvim/commit/71227a14435ce93997b3e79a04d8a8b761cfe6fa). But, in my own testing of the installation, I didn't find the `extraConfigLua` part to be necessary when using `grammarPackages` with  `nixGrammars`. That section would be relevant for user's not installing through `nixGrammars` and utilizing the `settings.ensure_installed`.

Reply:
Hmm. This might be because I'm still on the `24.05` branch. I could not get it to work without adding it into the runtime path like that.